### PR TITLE
raidemulator: Convert CombatantState to typescript

### PR DESCRIPTION
**Pull request by valarnin**
Looks like I was able to pull this conversion to its own PR after all.

PR 3 of 4 for #2936, I guess. Once the last PR is open, I'll close the old PR.

### CODE REVIEW

File: ui/raidboss/emulator/data/CombatantState.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    this.posZ = posZ;
+    this.heading = heading;
+    this.targetable = targetable;
+    this.HP = HP;
+    this.maxHP = maxHP;
+    this.MP = MP;
+    this.maxMP = maxMP;
+  }
+
+  partialClone(props: ICombatantState): CombatantState {
```
Comment (quisquous) [inline ui/raidboss/emulator/data/CombatantState.ts:63]:
```suggestion
  partialClone(props: Partial<CombatantState>): CombatantState {
```

`ICombatantState` looks like it's mostly repeating `CombatantState`.  What about using `Partial` and only having one type?

File: ui/raidboss/emulator/data/CombatantState.ts [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+  targetable?: boolean;
+  HP?: number;
+  maxHP?: number;
+  MP?: number;
+  maxMP?: number;
+}
+
+// Member names taken from OverlayPlugin's MiniParse.cs
+// Types taken from FFXIV parser plugin
+export interface IPluginState {
```
Comment (quisquous) [inline ui/raidboss/emulator/data/CombatantState.ts:15]: I have to admit I'm not a huge fan of `I` naming.  Could this be `PluginState` or `PluginCombatant`?

File: ui/raidboss/emulator/data/CombatantState.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+export default class CombatantState implements ICombatantState {
+  posX: number;
+  posY: number;
+  posZ: number;
+  heading: number;
+  targetable: boolean;
+  HP: number;
+  maxHP: number;
+  MP: number;
+  maxMP: number;
```
Comment (quisquous) [inline ui/raidboss/emulator/data/CombatantState.ts:47]:
```suggestion
  hp: number;
  maxHp: number;
  mp: number;
  maxMp: number;
```

Sorry again about the lack of lint rule.

**Comment by valarnin** [comment]
Whoops, made those changes on the wrong branch...

File: ui/raidboss/emulator/data/CombatantState.ts (hunk 3 above)
Comment (valarnin) [reply ui/raidboss/emulator/data/CombatantState.ts]: Had to update some other spots in code to reflect this change.

**Comment by quisquous** [comment]
Looks like this has a conflict, but I don't have any other comments.  Can you resolve that and then I'll merge this?

**Pull request merged**